ampproject/amphtml
# 🏗 Migrate `amphtml` package management from `yarn` to `npm`

This PR changes AMP's package manager from `yarn` to `npm`. For a detailed discussion of why, see #30518.

**PR highlights:**
- Replaces all `yarn.lock` files in the repo with `package-lock.json`
- Updates Travis and GH Actions config files to no longer install / use / cache `yarn`
- Updates `check-package-manager.js` to error out when `yarn` is used to install dependencies
- Replaces the `yarn` integrity check for a stale `node_modules` with [`check-dependencies`](https://www.npmjs.com/package/check-dependencies), which runs in O(milliseconds)
- Updates PR check jobs to error out in the presence of uncommitted changes to `package-lock.json`
- Updates `validator/` scripts to use `npm install` instead of `yarn`
- Updates all `amphtml` developer docs

**Tested the following workflows:**
- Clean install (works)
- In-place install after syncing branch (works)
- Clean / in-place install when required tools are missing (both work, with expected warnings + steps)
- Install on different OSs (lockfile stable across Linux, macOS, and Windows)
- Run `gulp` task without installing packages (prompted to run `npm install`, `yarn` used to error out)
- Spot check changes between lockfiles (caret semver [resolution bug](https://github.com/ampproject/amphtml/pull/30700#issue-504619950) present in `yarn` has been fixed by `npm`)
- CI checks (sent out #30700 to pre-emptively fix some lint errors)

Fixes #30518
